dist/common/scripts/scylla_util.py

```diff
@@ -73,6 +73,11 @@ def datadir_p():
 def scyllabindir_p():
     return scylladir_p() / 'bin'
 
+def sysconfdir_p():
+    p = scylladir_p() / 'SCYLLA-SYSCONFDIR-FILE'
```

Comment:
The `SCYLLA-SYSCONFDIR-FILE` is bit odd, I think. Would a symbolic link do the trick? @avikivity what do you think?

Reply:
It breaks compatibility with existing installations. We must be able to load existing /etc/sysconfig/scylla-server files.

Let's move the default location to /etc/scylla.d. See http://0pointer.de/blog/projects/on-etc-sysinit.html.

We already have 

```
EnvironmentFile=/etc/sysconfig/scylla-server
EnvironmentFile=/etc/scylla.d/*.conf
```

Let's change that to 

```
EnvironmentFile=-/etc/sysconfig/scylla-server
EnvironmentFile=-/etc/default/scylla-server
EnvironmentFile=/etc/scylla.d/*.conf
```

This way /etc/sysconfig and /etc/default are still supported, but optional. scylla_setup can write to files in /etc/scylla.d (it's simpler since each component can write to a different file).

Reply:
This does not trying to change sysconfig directory path, what the patch trying to do is allow python script to use --sysconfdir parameter passed to install.sh.

On our systemd unit file we generate sysconfdir.conf to customize EnvironmentFile directive as follows:
```
if [ "$sysconfdir" != "/etc/sysconfig" ]; then
    cat << EOS > "$retc"/systemd/system/scylla-server.service.d/sysconfdir.conf
[Service]
EnvironmentFile=
EnvironmentFile=$sysconfdir/scylla-server
EnvironmentFile=/etc/scylla.d/*.conf
EOS
fi
```

Unlike systemd unit file, current python script is hardcoded the path, so it's inconsistent:
```
if is_redhat_variant():
    cfg = sysconfig_parser('/etc/sysconfig/scylla-server')
else:
    cfg = sysconfig_parser('/etc/default/scylla-server')
```

If user specified non-standard path to install.sh, python script will fail (on Gentoo, sysconfig directory is /etc/conf.d)
The patch is adding a text file to store specified sysconfig directory path (SCYLLA-SYSCONFDIR-FILE), and read it from python script.

But I realized we don't really need the file, we can directly generate a simple python module file in install.sh instead, something like this:
```
 cat << EOS > "$rprefix"/scripts/scylla_sysconfdir.py
SYSCONFDIR="$sysconfdir"
EOS
```
Then use it from scylla_util.py:
```
--- a/dist/common/scripts/scylla_util.py
+++ b/dist/common/scripts/scylla_util.py
@@ -34,6 +34,7 @@ from pathlib import Path, PurePath
 from subprocess import run, DEVNULL
 
 import distro
+from scylla_sysconfdir import SYSCONFDIR
 
 
 def scriptsdir_p():
@@ -74,9 +75,7 @@ def scyllabindir_p():
     return scylladir_p() / 'bin'
 
 def sysconfdir_p():
-    p = scylladir_p() / 'SCYLLA-SYSCONFDIR-FILE'
-    with p.open('r', encoding='utf-8') as f:
-        return Path(f.read().strip())
+    return Path(SYSCONFDIR)
 
 def scriptsdir():
     return str(scriptsdir_p())
```


Moving sysconfig file to /etc/scylla.d sounds nice, easier to handle the file from script, but it may cause issue on upgrading from previous versions, I think it's safer to keep current path.

Reply:
Pushed suggested code (generating python module in install.sh)